laterpay/application/Controller/Admin/Account.php

```diff
@@ -264,4 +269,52 @@ protected static function update_plugin_mode() {
 
         die;
     }
+
+    /**
+     * Toggle test mode between invisible and visible.
+     *
+     * @return void
+     */
+    public static function change_test_mode() {
+        $plugin_test_mode_form = new LaterPay_Form_TestMode();
+
+        if ( ! $plugin_test_mode_form->is_valid( $_POST ) ) {
+            wp_send_json(
+                array(
+                    'success' => false,
+                    'message' => __( 'Error occurred. Incorrect data provided.', 'laterpay' )
+                )
+            );
+        }
+        $is_in_visible_test_mode = $plugin_test_mode_form->get_field_value( 'plugin_is_in_visible_test_mode' );
+        $is_credentials_invalid = $plugin_test_mode_form->get_field_value( 'invalid_credentials' );
+
+        if ( $is_credentials_invalid ) {
+            update_option( 'laterpay_is_in_visible_test_mode', 0 );
+            wp_send_json(
+                array(
+                    'success'   => false,
+                    'mode'      => 'test',
+                    'message'   => __( 'The LaterPay plugin needs valid API credentials to work.', 'laterpay' ),
+                )
+            );
+
+        }
+
+        update_option( 'laterpay_is_in_visible_test_mode', $is_in_visible_test_mode );
```

Comment:
@yastapenko-ScnSoft wouldn't this line just overwrite the update performed in L293?
As in: if my settings are invalid, but $is_in_visible_test_mode is 1, L293 sets option 'laterpay_is_in_visible_test_mode' to 0, but the code just continues and sets $is_in_visible_test_mode (which is 1) directly thereafter?